zk/conn.go

```diff
@@ -312,10 +312,7 @@ func WithMaxConnBufferSize(maxBufferSize int) connOption {
 func (c *Conn) Close() {
 	close(c.shouldQuit)
 
-	select {
-	case <-c.queueRequest(opClose, &closeRequest{}, &closeResponse{}, nil):
-	case <-time.After(time.Second):
-	}
+	<-c.queueRequestWithTimeout(opClose, &closeRequest{}, &closeResponse{}, nil, time.Second)
```

Comment:
returning channels, especially here is not usually the cleanest way. here, we can block inside the method and this call can just not have a blocking channel receiver.